h/api/models/test/document_test.py

```diff
@@ -119,6 +122,416 @@ def test_document_find_or_create_by_uris_no_results():
     assert docuri.type == 'self-claim'
 
 
+create_or_update_document_uri_fixtures = pytest.mark.usefixtures(
+    'DocumentURI',
+    'log',
+)
+
+
+def mock_docuri_dict(uri=None):
+    """Return a mock docuri_dict for create_or_update_document_uri()."""
+    if uri is None:
+        uri = 'http://example.com/example_uri.html'
+
+    now = datetime.datetime.now()
+
+    return {
+        'type': 'self-claim',
+        'claimant': 'http://example.com/example_claimant.html',
+        'uri': uri,
+        'created': now,
+        'updated': now
+    }
+
+
+def mock_document():
+    """Return a mock Document object for create_or_update_document_uri()."""
+    class Document:
+        @property
+        def id(self):
+            pass
+        @property
+        def created(self):
+            pass
+        @property
+        def updated(self):
+            pass
+    return mock.Mock(spec=Document)
+
+
+def mock_db():
+    """Return a mock db session object."""
+    class DB:
+        def add(self, obj):
+            pass
+    return mock.Mock(spec=DB())
+
+
+@create_or_update_document_uri_fixtures
+def test_create_or_update_document_uri_calls_filter(DocumentURI):
+    document.create_or_update_document_uri(
+        db=mock_db(),
+        claimant='http://example.com/example_claimant.html',
+        uri='http://example.com/example_uri.html',
+        type='self-claim',
+        content_type=None,
+        document=mock_document(),
+        created=datetime.datetime.now(),
+        updated=datetime.datetime.now())
+
+    # FIXME: We need to assert that this is called with the right arguments,
+    # but that's very awkward to do with the way sqlalchemy querying works.'
+    # Move this functionality onto the model class itself where the tests can
+    # use the actual database.
+    assert DocumentURI.query.filter.call_count == 1
```

Comment:
Like you said in the comment, I think testing the querying by creating a document with a document uri in the database and then making sure that this one is being used is probably the better way to test this.
What's the reason for the need to move this onto the model class itself to be able to use the database in the tests?